lib/elixir_make/artefact.ex

```diff
@@ -187,22 +216,48 @@ defmodule ElixirMake.Artefact do
   @doc """
   Returns the url for the current target.
   """
-  def current_target_url(config, precompiler, nif_version) do
+  def current_target_url(config, precompiler, current_nif_version) do
     case precompiler.current_target() do
       {:ok, current_target} ->
-        available_urls = available_target_urls(config, precompiler)
-        target_at_nif_version = {current_target, nif_version}
+        nif_versions =
+          config[:make_precompiler_nif_versions] ||
+            [versions: []]
+
+        versions = nif_versions[:versions]
+        fallback_version = nif_versions[:fallback_version]
+
+        nif_version_to_use =
+          case find_nif_version(current_nif_version, versions) do
+            :no_candidates ->
+              if is_function(fallback_version, 3) do
+                fallback_version.(current_target, current_nif_version, versions)
+              else
+                current_nif_version
+              end
+
+            version when is_binary(version) ->
+              version
+          end
```

Comment:
I was thinking more something like this:

```suggestion
        fallback_version = nif_versions[:fallback_version] || &find_nif_version/3
        nif_version_to_use = fallback_version.(current_target, current_nif_version, versions)
```

This way someone has full control over the algorithm. But maybe that's too flexible? Do you think someone would even want to customize the default behaviour?

Reply:
Ah, maybe I was overthinking regarding how users may want to use this feature. It should be already flexible enough for most cases. I'll apply the suggestion.